src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -355,7 +211,29 @@ internal partial class BoundIndexerAccess : IIndexedPropertyReferenceExpression
 
         ISymbol IMemberReferenceExpression.Member => this.Indexer;
 
-        ImmutableArray<IArgument> IHasArgumentsExpression.ArgumentsInEvaluationOrder => BoundCall.DeriveArguments(this.Arguments, this.ArgumentNamesOpt, this.ArgsToParamsOpt, this.ArgumentRefKindsOpt, this.Indexer.Parameters, this.Syntax);
+        ImmutableArray<IArgument> IHasArgumentsExpression.ArgumentsInEvaluationOrder
+        {
+            get
+            {
+                MethodSymbol accessor = this.IsLeftOfAssignment ? this.Indexer.SetMethod : this.Indexer.GetMethod; 
```

Comment:
>SetMethod [](start = 79, length = 9)

Is this the API used by LocalRewriter? If not, please use the same API and add a test to observe the difference.